### PR TITLE
Added CTest support for automated testing

### PR DESCRIPTION
**Pull request by ravil-mobile**
* generates and runs numerics tests: minimal, matmul
* tests TensorView
* allows to use either OpenBLAS or LIBXSMM
* Eigen is going to be added in the next commit

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+# find appropriate compiler flags
+if (${ARCH} STREQUAL "snb")
+    set(CPU_FLAGS "-mavx")
+elseif (${ARCH} STREQUAL "hsw")
+    set(CPU_FLAGS "-march=core-avx2")
+elseif(${ARCH} STREQUAL "skx")
+    set(CPU_FLAGS "-xCORE-AVX512" "-fma")
+elseif(${ARCH} STREQUAL "thunderx2t99")
+    set(CPU_FLAGS "-thunderx2t99")
+else()
```
Comment (krenzland) [inline tests/CMakeLists.txt:36]: That has to be wrong, right?

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    set(ARCH_STR s${ARCH})
+endif()
+
+# find appropriate compiler flags
+if (${ARCH} STREQUAL "snb")
+    set(CPU_FLAGS "-mavx")
+elseif (${ARCH} STREQUAL "hsw")
+    set(CPU_FLAGS "-march=core-avx2")
+elseif(${ARCH} STREQUAL "skx")
+    set(CPU_FLAGS "-xCORE-AVX512" "-fma")
```
Comment (krenzland) [inline tests/CMakeLists.txt:33]: These do not look optimal - couldn't we just use -mcpu=native -mtune=native (one of both is good enough, can't remember which one imples which) for unit tests as default?

File: tests/include [hunk 3]
```diff
@@ -0,0 +1 @@
+../include
```
Comment (krenzland) [inline tests/include:1]: Is there a reason why we would want this instead of adjusting the cmake include dir?

File: tests/CMakeLists.txt (hunk 1 above)
Comment (ravil-mobile) [reply tests/CMakeLists.txt:36]: right, it is a typo. I guess it should be
```
-mcpu=thunderx2t99
```

File: tests/CMakeLists.txt (hunk 2 above)
Comment (ravil-mobile) [reply tests/CMakeLists.txt:33]: I just took it from SeisSol/cmake/cpu_arch_flags.cmake. Let's make just
```
-mcpu=native
```
for all architectures. I guess a compiler should figure out which flags to pick up

Comment (krenzland) [reply tests/CMakeLists.txt]: I think this should be -mcpu=native for arm and -march=native for all other architectures. Please don't ask why, but -mcpu is a deprecated synonym for -mtune of x86 but this is not the case for ARM. And mcpu doesn't give the best performance, you'd need to use -march then.

> In short, you cannot support both x86 and Arm with exactly the same compiler flags.  You must use -mcpu for Arm and -march for x86.

See [here](https://community.arm.com/developer/tools-software/tools/b/tools-software-ides-blog/posts/compiler-flags-across-architectures-march-mtune-and-mcpu) for all the details you didn't want to know.

Sorry for being pedantic about this - the reason I am is that I also wanted to have =native settings for SeisSol. Still not sure what flags we'd need to set...

**Pull request merged**